docs/using_python_client_with_hazelcast.rst

```diff
@@ -1572,22 +1572,114 @@ Query parameters have the following benefits:
 <https://docs.hazelcast.com/hazelcast/latest/sql/create-mapping.html>`__
 statement, similar to the example above.
 
 For example, this code snippet creates a mapping to a new map called
-``json_employees``, which stores the JSON values ``name`` and ``salary`` and
-query it:
+``json_employees``, which stores the JSON values as ``HazelcastJsonValue``
+objects and queries it using nested fields, which is not possible with the
+``json-flat`` type:
 
 .. code:: python
 
-    client = hazelcast.HazelcastClient()
+    client.sql.execute(
+        """
+    CREATE MAPPING IF NOT EXISTS json_employees
+    TYPE IMap
+    OPTIONS (
+        'keyFormat' = 'int',
+        'valueFormat' = 'json'
+    )
+        """
+    ).result()
+
+    json_employees = client.get_map("json_employees").blocking()
+
+    json_employees.set(
+        1,
+        HazelcastJsonValue(
+            {
+                "personal": {"name": "John Doe"},
+                "job": {"salary": 60000},
+            }
+        ),
+    )
+
+    json_employees.set(
+        2,
+        HazelcastJsonValue(
+            {
+                "personal": {"name": "Jane Doe"},
+                "job": {"salary": 80000},
+            }
+        ),
+    )
+
+    with client.sql.execute(
+        """
+    SELECT JSON_VALUE(this, '$.personal.name') AS name
+    FROM json_employees
+    WHERE JSON_VALUE(this, '$.job.salary' RETURNING INT) > ?
+        """,
+        75000,
+    ).result() as result:
+        for row in result:
+            print(f"Name: {row['name']}")
+
+The ``json`` data type comes with full support for querying JSON in maps and
+Kafka topics.
+
+
+**JSON Functions**
+
+Hazelcast supports the following functions, which can retrieve JSON data.
+
+- `JSON_QUERY <https://docs.hazelcast.com/hazelcast/latest/sql/functions-and-operators#json_query>`__
+  : Extracts a JSON value from a JSON document or a JSON-formatted string that
+  matches a given JsonPath expression.
+
+- `JSON_VALUE <https://docs.hazelcast.com/hazelcast/latest/sql/functions-and-operators#json_value>`__
+  : Extracts a primitive value, such as a string, number, or boolean that
+  matches a given JsonPath expression. This function returns ``NULL`` if a
+  non-primitive value is matched, unless the ``ON ERROR`` behavior is changed.
+
+- `JSON_ARRAY <https://docs.hazelcast.com/hazelcast/latest/sql/functions-and-operators#json_array>`__
+  : Returns a JSON array from a list of input data.
+
+- `JSON_OBJECT <https://docs.hazelcast.com/hazelcast/latest/sql/functions-and-operators#json_object>`__
+  : Returns a JSON object from the given key/value pairs.
+
+**json-flat**
+
+To query ``json-flat`` objects, you should create an explicit mapping using the
+`CREATE MAPPING
+<https://docs.hazelcast.com/hazelcast/latest/sql/create-mapping.html>`__
+statement, similar to the example above.
+
+For example, this code snippet creates a mapping to a new map called
+``json_flat_employees``, which stores the JSON values with columns ``name``
```

Comment:
Do you think using just `employees` would make the example simpler?

Reply:
I wanted to distinguish the map/mapping names if the users are going to test the two code snippets. So, `json_employees` and `json_flat_employees` are coming from there, and I don't think it makes the example harder, as it is just a name